### ISSUE TITLE
Broken dependency chain during first image run when gem in `Gemfile` is not in cache

### ISSUE THREAD

**Issue by sgnn7** [puppetlabs/pdk-docker#12]
## Describe the Bug

Original bug filed [here](https://github.com/puppetlabs/puppet-module-gems/issues/154) until I pinpointed it to this repo

Effectively, if you are working on a puppet module and are:
- Running the `puppet/pdk` image for the first time
- or are missing the generated cache files (`.bundle/`, `.cache/`, `.config/`, `.pdk`, and `Gemfile.lock`)

and also have a gem in the `Gemfile` of your project that is not in baked-in gem cache, the image will exit with the following error:
```sh-session
13:08:46  An error occurred while installing http-parser (1.2.2), and Bundler cannot
13:08:46  continue.
13:08:46  Make sure that `gem install http-parser -v '1.2.2' --source
13:08:46  'https://rubygems.org/'` succeeds before bundling.
13:08:46  
13:08:46  In Gemfile:
13:08:46    puppet-module-posix-dev-r2.5 was resolved to 0.5.3, which depends on
13:08:46      puppet_litmus was resolved to 0.18.4, which depends on
13:08:46        rspec_honeycomb_formatter was resolved to 0.2.1, which depends on
13:08:46          honeycomb-beeline was resolved to 2.3.0, which depends on
13:08:46            libhoney was resolved to 1.15.0, which depends on
13:08:46              http was resolved to 4.4.1, which depends on
13:08:46                http-parser
```

This is because neither `gcc` nor `make` are installed in the image. 
After the first failure, the image works fine in the same folder.

## Expected Behavior
Run succeeds

### Easy reproduction:

1. Clone our project: https://github.com/cyberark/conjur-puppet
2. **You must have a clean checkout of this repo - problem is cache related**
3. Run `./scripts/test.sh`
4. **If you get a successful run, you must remove all the cache files to see the error again** 

### Minimal reproduction

- `Gemfile`
```gemfile
source 'https://rubygems.org'

group :development do
  gem "puppet-module-posix-dev-r2.5", '0.5.3',      require: false, platforms: [:ruby]
end
```
- `metadata.json`
```json
{
  "name": "cyberark-conjur",
  "version": "3.1.0",
  "dependencies": [
    {
      "name": "puppetlabs/registry",
      "version_requirement": ">= 3.1.0 < 4.0.0"
    }
  ],
  "operatingsystem_support": [
    {
      "operatingsystem": "Ubuntu",
      "operatingsystemrelease": [
        "20.04"
      ]
    }
  ],
  "requirements": [
    {
      "name": "puppet",
      "version_requirement": ">= 6.0.0 < 7.0.0"
    }
  ],
  "pdk-version": "1.18.1",
  "template-url": "pdk-default#1.18.1",
  "template-ref": "tags/1.18.1-0-g3d2e75c"
}
```

- Run  `docker run --rm -it -w /root -v $(pwd)/Gemfile:/root/Gemfile -v $(pwd)/metadata.json:/root/metadata.json puppet/pdk test unit`

## Environment
```
pdk (INFO): Using Ruby 2.5.8
pdk (INFO): Using Puppet 6.17.0
```
 - Platform: `puppet/pdk` DockerHub container 

## Additional Context

Since the breaking dependency is usually placed in the `:development` group, this is probably only visible for validation and test actions on the PDK image.

**Comment by DavidS** [comment]
we're waiting on a bolt release with https://github.com/puppetlabs/bolt/pull/2366/commits/bdc1bdca8012227bd2b5309f1656eb30063df2ac in it to update all the litmus deps to ruby 2.7 only. The release is expected for either today or tomorrow.

cc @lucywyman @npwalker

**Comment by DavidS** [comment]
I've been informed that there will be no bolt release in this week of the :turkey:.

**Comment by sgnn7** [comment]
Hey @DavidS,
It's not an urgency for us as we just opted to use [our own Ubuntu image with `pdk` installed](https://github.com/cyberark/conjur-puppet/blob/master/ci/Dockerfile.pdk) as a working alternative for now but thank you for the update!

**Comment by carabasdaniel** [comment]
Hi @sgnn7,  @DavidS 
Ran the tests with the nightly image of pdk and it seems that the issue has been fixed, we'll close the issue once the pdk image will be promoted to latest.

Test results from running with pdk nightly image:
`Running validations...
pdk (WARN): This module is compatible with an older version of PDK. Run `pdk update` to update it to your version of PDK.
pdk (INFO): Using Ruby 2.5.8
pdk (INFO): Using Puppet 6.19.1
pdk (INFO): Validator 'puppet-epp' skipped for '/root'. No files matching '["**/*.epp"]' found to validate.
Running specs...
pdk (WARN): This module is compatible with an older version of PDK. Run `pdk update` to update it to your version of PDK.
pdk (INFO): Using Ruby 2.5.8
pdk (INFO): Using Puppet 6.19.1
Run options: exclude {:bolt=>true}
  Evaluated 80 tests in 129.2321691 seconds: 0 failures, 0 pending.
Tests complete!
`

**Comment by lucywyman** [comment]
Huh I'm not sure how this flew under my radar, but we just shipped Bolt 3.0 with Ruby 2.7!

**Comment by sanfrancrisko** [comment]
Closing as this should now be resolved in `2.1.0.0`

**Comment by cdenneen** [comment]
Seems broken again

```
In Gemfile:
281
  puppet-module-posix-dev-r2.5 was resolved to 0.5.3, which depends on
282
    puppet_litmus was resolved to 0.28.0, which depends on
283
      bolt was resolved to 3.16.1, which depends on
284
        net-ssh-krb was resolved to 0.5.1, which depends on
285
          gssapi was resolved to 1.3.1, which depends on
286
            ffi
287
```
